shims/emr52/impl/pom.xml

```diff
@@ -314,36 +314,72 @@
       <artifactId>hadoop-mapreduce-client-app</artifactId>
       <version>${org.apache.hadoop.version}</version>
       <scope>compile</scope>
+      <exclusions>
```

Comment:
defining of avro dependency on top like you did in common-shim will redefine needed version, 
 <dependency>
 +      <groupId>org.apache.avro</groupId>
 +      <artifactId>avro</artifactId>
 +      <version>${org.apache.avro.version}</version>
 +      <!--<scope>provided</scope>-->
 +    </dependency>